cmd/embedded-cluster/uninstall.go

```diff
@@ -379,39 +400,43 @@ var resetCommand = &cli.Command{
 			}
 		}
 
-		// drain node
-		logrus.Info("Draining node...")
-		err = currentHost.drainNode()
-		if !checkErrPrompt(c, err) {
-			return err
-		}
-
-		// remove node from cluster
-		logrus.Info("Removing node from cluster...")
-		removeCtx, removeCancel := context.WithTimeout(c.Context, time.Minute)
-		defer removeCancel()
-		err = currentHost.deleteNode(removeCtx)
-		if !checkErrPrompt(c, err) {
-			return err
-		}
-
-		// controller pre-reset
-		if currentHost.Status.Role == "controller" {
-
-			// delete controlNode object from cluster
-			deleteControlCtx, deleteCancel := context.WithTimeout(c.Context, time.Minute)
-			defer deleteCancel()
-			err := currentHost.deleteControlNode(deleteControlCtx)
+		numControllerNodes, _ := kubeutils.NumOfControlPlaneNodes(c.Context, currentHost.Kclient)
+		// do not drain node if this is the only controller node in the cluster
+		// if there is an error (numControllerNodes == 0), drain anyway to be safe
+		if currentHost.Status.Role != "controller" || numControllerNodes != 1 {
```

Comment:
This is a really bad diff. I didnt change anything inside this conditional. I only wrapped the code in it.

Reply:
looks much better when ignoring whitespace